CHANGES

```diff
@@ -10,6 +10,12 @@ Dependencies
 Incompatible changes
 --------------------
 
+* #11459: ``sphinx.ext.autodoc.preserve_defaults.get_function_def`` is
+  renamed ``sphinx.ext.autodoc.preserve_defaults.get_arguments`` and its
+  signature is updated accordingly. It now extracts the ``ast.arguments`` from
+  a function or a lambda function object.
+  Patch by Bénédikt Tran.
```

Comment:
We can't have incompatible changes to public API in a minor release. Is there evidence of usage of this function in the wild?

Reply:
I don't really know but since the function name indicated a public name, I thought there might be occurrences. The issue with that function was that it did not do what it was supposed to do at first (for instance the return type was not correct in all cases). 

I can keep the old implementation and schedule it for removal if it's ok for you.